app/controllers/user_controller.rb

```diff
@@ -223,9 +223,9 @@ def signchangeemail
       params[:signchangeemail].delete(:password)
       post_redirect = PostRedirect.new(:uri => signchangeemail_url,
                                        :post_params => params,
+                                       :user => @user,
```

Comment:
Need user here so the `after_initialize` `#generate_email_token` method has access to it.